### ISSUE TITLE
Amino acids in gene call file translated in wrong direction

### ISSUE THREAD

**Issue by JBuongio** [merenlab/anvio#938]
(anvio-5.1) barbara-lloydlab@lloydbarbara:~$ anvi-self-test --version
Anvi'o version ...............................: margaret (v5.1-master)
Profile DB version ...........................: 29
Contigs DB version ...........................: 12
Pan DB version ...............................: 12
Genome data storage version ..................: 6
Auxiliary data storage version ...............: 2
Structure DB version .........................: 1
____________________________________________________________________________

Hi Meren, 

When using the --report-aa-seqs-for-gene-calls option in anvi-summarize, the translation from DNA to amino acid does not mind the direction of the DNA, and so many amino acid sequences are translated backward and have stop codons. 

Thanks for your help, always!!

**Comment by meren** [comment]
Hi @JBuongio,

Did you use external gene calls for this project?

Because the code seems to be ready to handle this unless the 'reverse' gene calls are not identified as 'reverse' in the contigs database. For future references the snippet from `summarizer.py`:

``` python
(...)

            # finally add the dna and amino acid sequence for gene calls:
            contig = self.summary.genes_in_contigs_dict[gene_callers_id]['contig']
            start = self.summary.genes_in_contigs_dict[gene_callers_id]['start']
            stop = self.summary.genes_in_contigs_dict[gene_callers_id]['stop']

            dna_sequence = self.summary.contig_sequences[contig]['sequence'][start:stop]
            if self.summary.genes_in_contigs_dict[gene_callers_id]['direction'] == 'r':
                dna_sequence = utils.rev_comp(dna_sequence)

            d[gene_callers_id]['dna_sequence'] = dna_sequence

            # if the user asked for it, report amino acid sequences as well
            if self.summary.report_aa_seqs_for_gene_calls:
                try:
                    d[gene_callers_id]['aa_sequence'] = utils.get_DNA_sequence_translated(dna_sequence, gene_callers_id)
                except:
                    d[gene_callers_id]['aa_sequence'] = ''

(...)
```

**Comment by JBuongio** [comment]
Yes, I imported prokka annotations as instructed with the gff parser.

**Comment by meren** [comment]
According to anvi'o Slack, and https://github.com/karkman/gff_parser/commit/232f1cc3308b4fc7d3802a17362ca6d02269c140, Antti solved this issue :)

**Comment by JBuongio** [comment]
Perfect!! Excited to use it  :) Thanks!

Joy
